docs/RFCS/replicate_queue_remove_replicas.md

```diff
@@ -0,0 +1,178 @@
+- Feature Name: Replicate Queue: Remove Replicas Enhancement
+- Status: draft
+- Start Date: 2015-08-18
+- RFC PR: #2153
+- Cockroach Issue: #620
+
+# Summary
+The Replicate Queue is currently used to add replicas to a range that is
+under-replicated. It can only add replicas, and it can only handle ranges with
+homogenous ReplicaAttrs (i.e. every replica requires the same attributes). This
+change will enhance the replicate queue to allow for down-replication: the
+removal of replicas from a range which has too many replicas. 
+
+With this addition, the replicate queue will be able to correct the replication
+of any range to its optimal replica count, regardless of how the range got into
+a sub-optimal state. Most immediately, this will provide crucial support for the
+highly desired "Rebalance" and "Repair" operations, but it will also provide
+support any other operation which needs to relocate a replica.
+
+This RFC will not address the issue of non-homogenous ReplicaAttrs. While not
+exceptionally difficult, we are simply not yet deploying with non-homogenous
+replicas and the work can be postponed.
+
+# Motivation
+Rebalancing is the major missing component from Cockroach at this point; without
+it, Cockroach is constrained to running on three stores. Rebalancing will
+opportunistically relocate replicas from over-utilized stores to under-utilized
+stores.
+
+Another important missing feature is Repair; detecting when a store is down, and
+relocating the replicas that were on that store. The relocation of these replicas
+is similar to Rebalancing, but happens in a slightly different fashion.
+
+Regardless of the origin, every relocation operation is conceptually the
+transfer of one replica in a range: the replica is removed from one store (the
+source) and added to another (the target).  To work with raft, this is
+explicitly divided into two separate operations: removing from the source store,
+and adding to the target store. These operations can generally happen in either
+order, but must be performed one at a time.
+
+The replicate queue can be responsible for the second operation of all
```

Comment:
I'm very confused as to why this is called the replicate queue. Perhaps you want to brief add another section at the top discussing it, or perhaps we need to rename it.